sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -75,13 +75,23 @@ public class AzurePipelinesCredential implements TokenCredential {
                     HttpRequest request = new HttpRequest(HttpMethod.POST, url);
                     request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                     request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
+                    request.setHeader(HttpHeaderName.fromString("X-TFS-FedAuthRedirect"), "Suppress");
                     try (HttpResponse response = httpPipeline.sendSync(request, Context.NONE)) {
                         String responseBody = response.getBodyAsBinaryData().toString();
                         if (response.getStatusCode() != 200) {
-                            throw LOGGER.logExceptionAsError(new ClientAuthenticationException("Failed to get the client assertion token "
+                            String xVssHeader = response.getHeaderValue(HttpHeaderName.fromString("x-vss-e2eid"));
```

Comment:
Just confirming, if the response doesn't contain the `x-vss-e2eid` header, the `getHeaderValue()` will just return null, correct, and not throw?